lib/JsonpMainTemplatePlugin.js

```diff
@@ -160,6 +169,17 @@ JsonpMainTemplatePlugin.prototype.apply = function(mainTemplate) {
 		}
 		return source;
 	});
+	mainTemplate.plugin("startup", function(source, chunk, hash) {
+		if(needChunkLoadingCode(chunk)) {
+			return this.asString([
+				"for(var i = 0; i < jsonpArray.length; i++) webpackJsonpCallback(jsonpArray[i]);",
+				"var parentJsonpFunction = jsonpArray.push;",
```

Comment:
This is not bound, so calling it will push into `undefined` or `global` (whichever is inappropriate 😄).

It also doesn't actually notify the parent that something changed.